src/frontend/src/common/components/library/data_table/index.tsx

```diff
@@ -112,13 +151,45 @@ export const DataTable: FunctionComponent<Props> = ({
   handleHeaderCheckboxClick,
   handleRowCheckboxClick,
   isHeaderChecked,
+  setIsHeaderChecked,
```

Comment:
I think it is a react anti pattern to allow a component to update another component's state in this way. Something feels weird about passing in the state modification functions.